processor/cloningfanoutconnector.go

```diff
@@ -150,7 +233,16 @@ func cloneTraceData(td *consumerdata.TraceData) *consumerdata.TraceData {
 	return clone
 }
 
-func cloneMetricsData(md *consumerdata.MetricsData) *consumerdata.MetricsData {
+func cloneTraceData(td data.TraceData) data.TraceData {
+	resourceSpansClones := make([]*otlptrace.ResourceSpans, 0, len(data.TraceDataToOtlp(td)))
+	for _, resourceSpans := range data.TraceDataToOtlp(td) {
```

Comment:
otlp := data.TraceDataToOtlp(td)

Just in case data.TraceDataToOtlp will become a bit more expensive.

Reply:
done